src/PSRule.Rules.Azure/rules/Azure.Defender.Rule.yaml

```
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#
+# Rules for Azure Defender
+#
+
+#region Rules
+
+---
+# Synopsis: 
+apiVersion: github.com/microsoft/PSRule/v1
+kind: Rule
+metadata:
+  name: Azure.Defender.Containers
+  ref: AZR-000289
+  tags:
+    release: 'GA'
+    ruleSet: '2022_09'
+spec:
+  level: Error
+  type:
+  - Microsoft.Security/pricings
+  condition:
+    allOf:
+      - name: '.'
+        equals: 'Containers'
+      - field: 'properties.pricingTier'
+        equals: 'Standard'
```

@jdewisscher Nice. You could make this even better by moving the name check to the pre-condition.

The main difference is:

- If the name check is in the main `condition` block, the rule will fail when other pricing plans exist such as `SqlServers`.
- If you put it in the `where` block, the rule will be skipped for other pricing plans, and will run for `Containers` pricing plans. When run:
  - If the `properties.pricingTier` property is `Standard` the rule will pass.
  - If the property does not exist the rule will fail.
  - If the property is any other value except `Standard` the rule will fail.

For example:

```suggestion
  where:
    name: '.'
    equals: Containers
  type:
  - Microsoft.Security/pricings
  condition:
    allOf:
      - field: 'properties.pricingTier'
        equals: 'Standard'
```

---

Excellent point, added!